FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,59 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			// to prevent memory leak
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, response)):
+				completion(FeedImagesMapper.map(data,response))
+
+			default:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+	
+	private final class FeedImagesMapper {
+		private struct Root: Decodable {
+			var items: [Image]
+
+			var feedImages: [FeedImage] {
+				return items.map { $0.feedImage }
+			}
+		}
+
+		private struct Image: Decodable {
+			let id: UUID
+			let imageDesc: String?
+			let imageLoc: String?
+			let imageURL: URL
+
+			enum CodingKeys: String, CodingKey {
+				case id = "image_id"
+				case imageDesc = "image_desc"
+				case imageLoc = "image_loc"
+				case imageURL = "image_url"
+			}
+
+			var feedImage: FeedImage {
+				return FeedImage(id: id, description: imageDesc, location: imageLoc, url: imageURL)
+			}
+		}
+		
+		static func map(_ data: Data, _ response: HTTPURLResponse) -> FeedLoader.Result {
+			guard response.statusCode == 200 else {
+				return .failure(Error.invalidData)
+			}
+			
+			if let root = try? JSONDecoder().decode(Root.self, from: data) {
+				return .success(root.feedImages)
+			} else {
+				return .failure(Error.invalidData)
+			}
```

Comment:
Both failure cases can be combined in one `guard ...` statement to remove duplication
```suggestion
			guard response.statusCode == 200, let root = try? JSONDecoder().decode(Root.self, from: data) else {
				return .failure(Error.invalidData)
			}
			
			return .success(root.feedImages)			
```